src/uwtools/tests/drivers/test_cdeps.py

```diff
@@ -0,0 +1,146 @@
+# pylint: disable=missing-function-docstring,protected-access,redefined-outer-name
+"""
+CDEPS driver tests.
+"""
+
+import datetime as dt
+import logging
+from copy import deepcopy
+from pathlib import Path
+from textwrap import dedent
+from unittest.mock import patch
+
+import f90nml  # type: ignore
+from iotaa import refs
+from pytest import fixture, mark
+
+from uwtools.config.formats.nml import NMLConfig
+from uwtools.drivers import cdeps
+from uwtools.drivers.cdeps import CDEPS
+from uwtools.logging import log
+from uwtools.tests.support import logged
+from uwtools.tests.test_schemas import CDEPS_CONFIG
+
+# Fixtures
+
+
+@fixture
+def driverobj(tmp_path):
+    return CDEPS(
+        config={"cdeps": {**deepcopy(CDEPS_CONFIG), "rundir": str(tmp_path)}},
+        cycle=dt.datetime.now(),
+    )
+
+
+# Tests
+
+
+def test_CDEPS_atm(driverobj):
+    with patch.object(CDEPS, "atm_nml") as atm_nml:
+        with patch.object(CDEPS, "atm_stream") as atm_stream:
+            driverobj.atm()
+        atm_stream.assert_called_once_with()
+    atm_nml.assert_called_once_with()
+
+
+@mark.parametrize("group", ["atm", "ocn"])
+def test_CDEPS_nml(caplog, driverobj, group):
+    log.setLevel(logging.DEBUG)
+    dst = driverobj._rundir / f"d{group}_in"
+    assert not dst.is_file()
+    del driverobj._driver_config[f"{group}_in"]["base_file"]
+    task = getattr(driverobj, f"{group}_nml")
+    path = Path(refs(task()))
+    assert dst.is_file()
+    assert logged(caplog, f"Wrote config to {path}")
+    assert isinstance(f90nml.read(dst), f90nml.Namelist)
+
+
+def test_CDEPS_ocn(driverobj):
+    with patch.object(CDEPS, "ocn_nml") as ocn_nml:
+        with patch.object(CDEPS, "ocn_stream") as ocn_stream:
+            driverobj.ocn()
+        ocn_stream.assert_called_once_with()
+    ocn_nml.assert_called_once_with()
+
+
+def test_CDEPS_provisioned_rundir(driverobj):
+    with patch.object(CDEPS, "atm") as atm:
+        with patch.object(CDEPS, "ocn") as ocn:
+            driverobj.provisioned_rundir()
+        ocn.assert_called_once_with()
+    atm.assert_called_once_with()
+
+
+@mark.parametrize("group", ["atm", "ocn"])
+def test_CDEP_streams(driverobj, group):
+    dst = driverobj._rundir / f"d{group}.streams"
+    assert not dst.is_file()
+    template = """
+    {{ stream01.dtlimit }}
+    {{ stream01.mapalgo }}
+    {{ stream01.readmode }}
+    {{ " ".join(stream01.stream_data_files) }}
+    {{ " ".join(stream01.stream_data_variables) }}
+    {{ stream01.stream_lev_dimname }}
+    {{ stream01.stream_mesh_file }}
+    {{ stream01.stream_offset }}
+    {{ " ".join(stream01.stream_vectors) }}
+    {{ stream01.taxmode }}
+    {{ stream01.tinterpalgo }}
+    {{ stream01.yearAlign }}
+    {{ stream01.yearFirst }}
+    {{ stream01.yearLast }}
+    """
```

Comment:
This test creates a simple Jinja2 template file that uses the values from the config to render to an output file, which is then checked against expected values.

Reply:
@maddenp-noaa Okay. I was wondering about the Jinja2 template. So, users could just generate it by running tests.

Reply:
For other drivers where we must currently rely on a Jinja2 template, we're expecting that the Apps that make use of `uwtools` will provide the appropriate template file(s) in their repo. Then the UW YAML for the component (CDEPS in this case) can point to that template file in the App's git clone (for example) as the source.